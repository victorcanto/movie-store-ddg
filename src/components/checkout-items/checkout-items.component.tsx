import { Box, Grid, List, Typography, useTheme } from '@mui/material';
import * as React from 'react';
import { CartItem, removeCartItem } from '../../shared';
import {
	selectCartItems,
	setCartItems,
	useAppDispatch,
	useAppSelector,
} from '../../store';
import { ListItem } from '../list-item/list-item.component';

export function CheckoutItems() {
	const theme = useTheme();
	const dispatch = useAppDispatch();
	const cartItems = useAppSelector(selectCartItems);

	const removeItemHander = (item: CartItem) => () => {
		const newCartItems = removeCartItem(cartItems, item);
		if (newCartItems) {
			dispatch(setCartItems(newCartItems));
		}
	};

	return (
		<Box
			sx={{
				width: '100%',
				height: '100%',
			}}
		>
			<Grid data-testid='checkout-items' container spacing={2}>
				<Grid item xs={12} md={12}>
					<List>
						{cartItems.map((item) => {
							return (
								<ListItem
									key={item.id}
									item={item}
									removeItem={removeItemHander(item)}
								/>
							);
						})}
					</List>
				</Grid>
			</Grid>
		</Box>
	);
}
