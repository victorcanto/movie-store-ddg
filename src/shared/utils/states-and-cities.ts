import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
