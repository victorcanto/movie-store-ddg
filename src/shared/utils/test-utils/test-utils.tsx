import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
