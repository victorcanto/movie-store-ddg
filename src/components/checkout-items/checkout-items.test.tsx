import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { CheckoutItems } from './checkout-items.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<CheckoutItems />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId } = render(<CheckoutItems />);
	expect(getByTestId('checkout-items')).toBeInTheDocument();
});
