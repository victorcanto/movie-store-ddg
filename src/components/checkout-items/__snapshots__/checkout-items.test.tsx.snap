// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-uwwqev"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      data-testid="checkout-items"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 css-1idn90j-MuiGrid-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
